map/src/main/kotlin/edu/artic/map/MapViewModel.kt

```diff
@@ -56,22 +57,35 @@ class MapViewModel @Inject constructor(
         }
 
     init {
-        val allAmenities = mapAnnotationDao
-                .getAmenitiesOnMap()
-                .map { annotationList ->
-                    annotationList.mapToMapItem()
-                }
 
-        val allBuildingNames = mapAnnotationDao
-                .getBuildingNamesOnMap()
-                .map { annotationList ->
-                    annotationList.mapToMapItem()
-                }
+        // Each time the floor changes update the current amenities for that floor this is explore mode
+        floor.distinctUntilChanged()
+                .flatMap { floor ->
+                    mapAnnotationDao.getAmenitiesOnMapForFloor(floor.toString()).asObservable()
+                }.map { amenitiesList -> amenitiesList.mapToMapItem() }
+                .bindTo(amenities)
+                .disposedBy(disposeBag)
+
 
-        Observables.combineLatest(allAmenities.asObservable(), allBuildingNames.asObservable())
-        { amenities, buildingNames ->
-            return@combineLatest amenities.toMutableList() + buildingNames
-        }.bindTo(alwaysVisibleAnnotations)
+        /**
+         * Upon changing zoom level to anything except MapZoomLevel.One and any distinct change in
+         * floor causes a change of space landmarks
+         */
+        Observables.combineLatest(
+                zoomLevel.distinctUntilChanged()
+                        .filter { zoomLevel -> zoomLevel !== MapZoomLevel.One },
+                floor.distinctUntilChanged()
+        ) { _, floor ->
+            mapAnnotationDao.getTextAnnotationByTypeAndFloor(
+                    ArticMapTextType.SPACE,
+                    floor.toString())
+                    .asObservable()
+                    .map { it.mapToMapItem() }
+        }
+                .flatMap {
+                    it
```

Comment:
don't think this is needed now?

Reply:
yea it is. for some reason otherwise it returns an observable<...> this is the only way I can think of to unwrap it @agrosner